pkg/meta/spec.go

```diff
@@ -0,0 +1,152 @@
+package meta
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/joomcode/errorx"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiup/pkg/cliutil"
+	"github.com/pingcap/tiup/pkg/file"
+	"github.com/pingcap/tiup/pkg/utils"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	errNS        = errorx.NewNamespace("spec")
+	errNSCluster = errNS.NewSubNamespace("cluster")
+	// ErrClusterCreateDirFailed is ErrClusterCreateDirFailed
+	ErrClusterCreateDirFailed = errNSCluster.NewType("create_dir_failed")
+	// ErrClusterSaveMetaFailed is ErrClusterSaveMetaFailed
+	ErrClusterSaveMetaFailed = errNSCluster.NewType("save_meta_failed")
+)
+
+const (
+	// MetaFileName is the file name of the meta file.
+	MetaFileName = "meta.yaml"
+	// PatchDirName is the directory to store patch file eg. {PatchDirName}/tidb-hotfix.tar.gz
+	PatchDirName = "patch"
+	// BackupDirName is the directory to save backup files.
+	BackupDirName = "backup"
+)
+
+// Spec control management of spec meta data.
+type Spec struct {
+	base string
+}
+
+// NewSpec create a spec instance.
+func NewSpec(base string) *Spec {
+	return &Spec{
+		base: base,
+	}
+}
+
+// ClusterPath returns the full path to a subpath (file or directory) of a
+// cluster, it is a subdir in the profile dir of the user, with the cluster name
+// as its name.
+func (s *Spec) ClusterPath(cluster string, subpath ...string) string {
```

Comment:
It's a good idea to abstract a new type to organize metadata, I think all exported methods name should refactor and shouldn't contain the `Cluster` .